File: pages/base_page.py
from playwright.sync_api import Page, expect


class BasePage:
    def __init__(self, page: Page):
        self.page = page

    def navigate(self, path: str = ""):
        self.page.goto(path)

    def get_title(self) -> str:
        return self.page.title()

    def get_url(self) -> str:
        return self.page.url

    def take_screenshot(self, name: str) -> bytes:
        return self.page.screenshot(path=f"reports/{name}.png")

    def wait_for_element(self, selector: str, timeout: int = 10000):
        self.page.wait_for_selector(selector, timeout=timeout)

    def is_visible(self, selector: str) -> bool:
        return self.page.is_visible(selector)

File: src/pages/BasePage.ts
import { Page } from '@playwright/test';
import { DEMO_PAUSE_MS } from '@utils/config';

export class BasePage {
  constructor(protected page: Page) {}

  async navigate(url: string): Promise<void> {
    await this.page.goto(url);
    // Brief pause so the page is visible before the next action
    await this.page.waitForTimeout(DEMO_PAUSE_MS);
  }

  async pause(): Promise<void> {
    await this.page.waitForTimeout(DEMO_PAUSE_MS);
  }

  getTitle(): Promise<string> {
    return this.page.title();
  }

  getUrl(): string {
    return this.page.url();
  }
}
